### PR TITLE
remove auth and other headers from api configuration

### PR DESCRIPTION
**Pull request by paritosh-08**
This PR removes auth and other headers from the API Configuration struct. This PR also adds some documentation around how we can add headers to be sent in API requests.

### CODE REVIEW

File: ndc-client/src/apis/configuration.rs [hunk 1]
```diff
@@ -1,11 +1,14 @@
+use reqwest::header::{HeaderMap, HeaderValue};
+
 #[derive(Debug, Clone)]
 pub struct Configuration {
     pub base_path: String,
     pub user_agent: Option<String>,
     pub client: reqwest::Client,
```
Comment (paf31) [inline ndc-client/src/apis/configuration.rs:7]: `client` provides a way to edit the default set of headers. Why not use that?

File: ndc-client/src/apis/configuration.rs [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
     pub basic_auth: Option<BasicAuth>,
     pub oauth_access_token: Option<String>,
-    pub bearer_access_token: Option<String>,
+    // TODO: should we remove oauth access tokens and basic_auth as well?
```
Comment (paf31) [inline ndc-client/src/apis/configuration.rs:10]: Yes I think so.

File: ndc-client/src/apis/configuration.rs (hunk 1 above)
Comment (paritosh-08) [reply ndc-client/src/apis/configuration.rs:7]: Are you talking about [`ClientBuilder`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html)?

So, the Configuration will have the client (already configured with default headers). While initializing, the user will have to make sure that the appropriate headers are already part of the client.

I am okay with this change, the only downside is that while initializing, we will have to create a [`ClientBuilder`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html) first, then [add default headers](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.default_headers) and then create client (and handle errors as the [`ClientBuilder.build`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.build) returns `Result<Client>`).

File: ndc-client/src/apis/configuration.rs (hunk 2 above)
Comment (paritosh-08) [reply ndc-client/src/apis/configuration.rs]: Done

**Review by paf31 — APPROVED** [review]

**Comment by paritosh-08** [comment]
@paf31 I have removed the headers.

I have also added some documentation around how the client is supposed to be built with some default set of headers. Can you please review that as well?

File: ndc-client/src/apis/configuration.rs [hunk 3]
```diff
@@ -1,11 +1,42 @@
+/// Configuration for the API.
```
Comment (paf31) [inline ndc-client/src/apis/configuration.rs:1]:
```suggestion
/// Configuration for the API client
```

File: ndc-client/src/apis/configuration.rs [hunk 4]
```diff
@@ -1,11 +1,42 @@
...
+/// 
 #[derive(Debug, Clone)]
 pub struct Configuration {
     pub base_path: String,
     pub user_agent: Option<String>,
     pub client: reqwest::Client,
-    pub basic_auth: Option<BasicAuth>,
-    pub oauth_access_token: Option<String>,
-    pub bearer_access_token: Option<String>,
     pub api_key: Option<ApiKey>,
```
Comment (paf31) [inline ndc-client/src/apis/configuration.rs:40]: Let's remove this one too while we're at it, please.

**Review by paf31 — APPROVED** [review]
Looks good, please feel free to merge after removing `api_key` and `ApiKey`.

File: ndc-client/src/apis/configuration.rs (hunk 3 above)
Comment (paritosh-08) [reply ndc-client/src/apis/configuration.rs]: Thanks, done

File: ndc-client/src/apis/configuration.rs (hunk 4 above)
Comment (paritosh-08) [reply ndc-client/src/apis/configuration.rs]: Done

**Review by paf31 — APPROVED** [review]
Thanks!

**Pull request merged**